Firestore/core/src/core/sync_engine.cc

```diff
@@ -168,7 +168,8 @@ void SyncEngine::StopListening(const Query& query) {
 
   TargetId target_id = query_view->target_id();
   auto& queries = queries_by_target_[target_id];
-  queries.erase(std::remove(queries.begin(), queries.end(), query));
+  queries.erase(std::remove(queries.begin(), queries.end(), query),
+                queries.end());
```

Comment:
This is the reference for this fix: https://clang.llvm.org/extra/clang-tidy/checks/bugprone-inaccurate-erase.html

Reply:
I presume there would normally (perhaps always) only be one query to remove, so this code would work equivalently to the correct code, but this is a very good find regardless.